src/mongocxx/collection.cpp

```diff
@@ -178,7 +178,9 @@ mongocxx::stdx::optional<bsoncxx::document::value> find_and_modify(
 
     bsoncxx::document::view reply_view = reply.view();
 
-    if (reply_view["value"].type() == bsoncxx::type::k_null) {
+    const auto value = reply_view["value"];
+
+    if (!value || value.type() == bsoncxx::type::k_null) {
```

Comment:
Nice catch. I filed CXX-2539 to track this bug. I think it is worth noting in release notes, and may help with discoverability if users encounter this bug.